### PR TITLE
Feature intent interface

### PR DESCRIPTION
**Pull request by LipuFei**

### CODE REVIEW

File: resources/qml/LabelBar.qml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        {
+            id: repeater
+            model: base.model
+
+            Item
+            {
+                Layout.fillWidth: true
+                Layout.maximumWidth: index + 1 === repeater.count || repeater.count <= 1 ? itemSize : base.width / (repeater.count - 1)
+                height: label.height
+                Label
```
Comment (LipuFei) [inline resources/qml/LabelBar.qml]: The `Label` is missing `renderType: Text.NativeRendering`

File: resources/qml/LabelBar.qml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    id: base
+    property var model: null
+    property string modelKey: ""
+    property int itemSize: 14
+    height: childrenRect.height
+    RowLayout
+    {
+        anchors.left: parent.left
+        anchors.right: parent.right
+        height: label.height
```
Comment (LipuFei) [inline resources/qml/LabelBar.qml]: I got this QML error: `resources/qml/LabelBar.qml:20: ReferenceError: label is not defined`

File: resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml [hunk 3]
```diff
@@ -47,33 +35,134 @@ Item
 
         Label
         {
-            id: intentLabel
+            id: profileLabel
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml]: Same as above, this `Label` is missing `renderType: Text.NativeRendering`.

File: resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml [hunk 4]
```diff
@@ -47,33 +35,134 @@ Item
...
             anchors.right: parent.right
-            style: UM.Theme.styles.print_setup_header_button
-            activeFocusOnPress: true
+            width: UM.Theme.getSize("print_setup_big_item").width
+            height: textLabel.contentHeight + 2 * UM.Theme.getSize("narrow_margin").height
+            hoverEnabled: true
+
+            contentItem: Label
+            {
+                id: textLabel
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml]: Same as above, this `Label` is missing `renderType: Text.NativeRendering`.

File: resources/qml/PrintSetupSelector/Custom/MenuButton.qml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        if(index >= 0)
+        {
+            txt = txt.replace(txt.substr(index, 2), ("<u>" + txt.substr(index + 1, 1) + "</u>"))
+        }
+        return txt
+    }
+
+    contentItem: Label
+    {
+        id: textLabel
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/MenuButton.qml]: Same as above, this `Label` is missing `renderType: Text.NativeRendering`

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+                height: childrenRect.height
+                anchors
+                {
+                    left: parent.left
+                    right: parent.right
+                }
+
+                Label
+                {
+                    id: headerLabel
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Missing `renderType: Text.NativeRendering`

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+            {
+                width: manageProfilesButton.width
+                Label
+                {
+                    id: textLabel
+                    text: manageProfilesButton.text
+                    height: contentHeight
+                    anchors.left: parent.left
+                    anchors.leftMargin: UM.Theme.getSize("default_margin").width + UM.Theme.getSize("narrow_margin").width
+                    verticalAlignment: Text.AlignVCenter
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Missing `renderType: Text.NativeRendering`

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+                }
+                Label
+                {
+                    id: shortcutLabel
+                    text: Cura.Actions.manageProfiles.shortcut
+                    height: contentHeight
+                    anchors.right: parent.right
+                    anchors.rightMargin: UM.Theme.getSize("default_margin").width
+                    verticalAlignment: Text.AlignVCenter
```
Comment (LipuFei) [inline resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Missing `renderType: Text.NativeRendering`

File: resources/qml/RadioCheckbar.qml [hunk 9]
```diff
@@ -0,0 +1,152 @@
+import QtQuick 2.0
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.3
+
+Item
+{
+    id: base
+    property ButtonGroup buttonGroup: null
+
+    property color activeColor: "#3282ff"
```
Comment (LipuFei) [inline resources/qml/RadioCheckbar.qml]: Do those sizes and colors need to be themed?

Comment (nallath) [reply resources/qml/RadioCheckbar.qml]: Yup, will fix those.

File: resources/qml/LabelBar.qml (hunk 1 above)
Comment (LipuFei) [reply resources/qml/LabelBar.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml (hunk 3 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml (hunk 4 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/CustomPrintSetup.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/MenuButton.qml (hunk 5 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/MenuButton.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml (hunk 6 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml (hunk 7 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Fixed in 0926c223b

File: resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml (hunk 8 above)
Comment (LipuFei) [reply resources/qml/PrintSetupSelector/Custom/QualitiesWithIntentMenu.qml]: Fixed in 0926c223b

**Pull request merged**